lib/open3.rb

```diff
@@ -1116,49 +1143,59 @@ def pipeline_w(*cmds, &block)
   end
   module_function :pipeline_w
 
-  # Open3.pipeline_start starts a list of commands as a pipeline.
-  # No pipes are created for stdin of the first command and
-  # stdout of the last command.
+  # :call-seq:
+  #   Open3.pipeline_start([env, ] *cmds, options = {}) -> [wait_threads]
   #
-  #   Open3.pipeline_start(cmd1, cmd2, ... [, opts]) {|wait_threads|
-  #     ...
-  #   }
+  # Basically a wrapper for
+  # {Process.spawn}[https://docs.ruby-lang.org/en/master/Process.html#method-c-spawn]
+  # that:
   #
-  #   wait_threads = Open3.pipeline_start(cmd1, cmd2, ... [, opts])
-  #   ...
+  # - Creates a child process for each of the given +cmds+
+  #   by calling Process.spawn.
+  # - Waits for all child processes to exit.
   #
-  # Each cmd is a string or an array.
-  # If it is an array, the elements are passed to Process.spawn.
+  # With no block given, returns an array of the wait threads0
+  # for all of the child processes.
   #
-  #   cmd:
-  #     commandline                              command line string which is passed to a shell
-  #     [env, commandline, opts]                 command line string which is passed to a shell
-  #     [env, cmdname, arg1, ..., opts]          command name and one or more arguments (no shell)
-  #     [env, [cmdname, argv0], arg1, ..., opts] command name and arguments including argv[0] (no shell)
+  # Example:
   #
-  #   Note that env and opts are optional, as for Process.spawn.
+  #   Open3.pipeline_start("sort", "cat -n")
```

Comment:
We should have better examples because if arguments or EOF is not passed into `sort`, it hangs indefinitely. I don't think it's a good idea to show an example of an incorrect usage.

Reply:
Fixed.